src/app/signup/_components/EmailPasswordBox.tsx

```diff
@@ -5,28 +5,43 @@ import { Input } from "@/components/ui/input";
 import { Label } from "@/components/ui/label";
 import { cn } from "@/utils/tailwind-merge";
 import * as React from "react";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { z } from "zod";
+import { type User } from "../interfaces";
 
 interface EmailPasswordBoxProps {
   setValid: (valid: boolean) => void;
   formRef: React.RefObject<HTMLFormElement>;
+  data: User;
+  emailNotice: string;
+  setEmailNotice: (emailNotice: string) => void;
 }
 
 const emailSchema = z.string().email();
 
 export default function EmailPasswordBox(props: EmailPasswordBoxProps) {
-  const { setValid, formRef } = props;
-
-  const [emailNotice, setEmailNotice] = useState<string>("");
+  const { setValid, formRef, data, emailNotice, setEmailNotice } = props;
+  const [emailText, setEmailText] = useState(data.Email ? data.Email : "");
+  const [passwordText, setPasswordText] = useState(
+    data.Password ? data.Password : "",
+  );
+  const [cfPasswordText, setCfPasswordText] = useState(
+    data.Password ? data.Password : "",
+  );
 
   const [passwordNotice, setPasswordNotice] = useState<string>("");
 
   const [confirmPasswordNotice, setConfirmPasswordNotice] =
     useState<string>("");
 
+  useEffect(() => {
+    formCheck();
+  });
+
```

Comment:
ทำแบบนี้มันจะยิง request แบบเละเทะ เพราะมันรันทุกครั้งที่ render component ใหม่
อันนี้สำคัญมากๆ อย่าหาทำ ให้ใส่ dependencies บอกว่ารันตอนอันไหนเปลี่ยน
```suggestion
  useEffect(() => {
    formCheck();
  }, [*DEPENDENCIES*]);
```